pkg/sink/sink.go

```diff
@@ -65,6 +67,7 @@ type Sink struct {
 	// WGProcessTriggers keeps track of triggers or triggerGroups currently being processed
 	// Currently only used in tests to wait for all triggers to finish processing
 	WGProcessTriggers *sync.WaitGroup
+	ContextForEvents  context.Context
```

Comment:
Minor: It seems like we are creating an eventRecorder, associating it to a context and then storing it in a field in Sink? Wouldn't it be simpler store the eventRecorder in the Sink struct and pass it around? It seems like the only use for the contextForEvents is to extract the recorder from it in the events.Emit function - could we just directly pass around the EventRecorder?

Reply:
Modified :+1: